src/main/kotlin/com/strumenta/kolasu/model/Processing.kt

```diff
@@ -117,49 +110,22 @@ fun Node.processProperties(
 /**
  * @return the first node in the AST for which the predicate is true. Null if none are found.
  */
-fun Node.find(predicate: (Node) -> Boolean): Node? {
-    if (predicate(this)) {
-        return this
-    }
-    containmentProperties.forEach { p ->
-        when (val v = p.get(this)) {
-            is Node -> {
-                val res = v.find(predicate)
-                if (res != null) {
-                    return res
-                }
-            }
-            is Collection<*> -> v.forEach {
-                (it as? Node)?.let {
-                    val res = it.find(predicate)
-                    if (res != null) {
-                        return res
-                    }
-                }
-            }
-        }
-    }
-    return null
+fun Node.find(predicate: (Node) -> Boolean, walker: KFunction1<Node, Sequence<Node>> = Node::walk): Node? {
```

Comment:
Another one made nicer and more readable!
I would comment walker also here, as it is not super intuitive

Reply:
Also done.